### PR TITLE
feat: enable emojis as channel icons

### LINKED ISSUE #4809: Support assigning an emoji to a channel
Screen Shot 2022-02-14 at 7 31 26 PM

### Description

note: this should already be supported in status-go

- [ ] when creating/editing a channel, it should be possible to select an emoji, this can use the same component as the emoji selector
- [ ] the UI component for the channel should be modified so it displays the emoji

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #4809

StatusQ PR https://github.com/status-im/StatusQ/pull/566

image

Placeholder input:
image
The actual input will be added in another PR

### CODE REVIEW

**Comment by Khushboo-dev-cpp** [comment]
Hey @jrainville ! thanks for this will help me adapt for wallet cause I haven't started yet lol
however




This is the expected view right?
Looks different from the screenshot you have attached

**Comment by jrainville** [comment]
> This one seems to have transparent Emojis, I dont think that possible though?

@Khushboo-dev-cpp not sure what you mean. The is the "yam" (potato) emoji above that doesn't have a background.  Or is it something else? The size?

**Comment by Khushboo-dev-cpp** [comment]
> > This one seems to have transparent Emojis, I dont think that possible though?
> 
> @Khushboo-dev-cpp not sure what you mean. The is the "yam" (potato) emoji above that doesn't have a background. Or is it something else? The size?

Hey yes sorry I missed seeing that. It is already transparent. its all okay :)

File: ui/app/AppLayouts/Onboarding/views/LoginView.qml [hunk 1]
```diff
@@ -48,7 +48,7 @@ Item {
     }
 
     Component.onCompleted: {
-        resetLogin()
+        doLogin("111111")
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Onboarding/views/LoginView.qml:51]: missed to remove? and I know your fav password now :D lol

File: ui/app/AppLayouts/Chat/views/CommunityColumnView.qml [hunk 2]
```diff
@@ -89,7 +90,7 @@ Item {
     Loader {
         id: membershipRequests
 
-        property int nbRequests: root.communityData.pendingRequestsToJoin.count
+        property int nbRequests: root.communityData.pendingRequestsToJoin.count || 0
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Chat/views/CommunityColumnView.qml:93]: why is this needed ?

File: ui/imports/shared/status/StatusChatInput.qml [hunk 3]
```diff
@@ -1139,7 +1142,15 @@ Rectangle {
                 icon.name: "emojis"
                 type: StatusQ.StatusFlatRoundButton.Type.Tertiary
                 color: "transparent"
-                onClicked: togglePopup(emojiPopup, emojiBtn)
+                onClicked: {
+                    control.emojiPopupOpened = true
+                    // TODO fix clicking on the button when the popup is opened reopens it because the click outside closes it
+                    // emojiPopup.parent = control
+                    // emojiPopup.closePolicy = Popup.CloseOnEscape | Popup.CloseOnPressOutsideParent
```
Comment (Khushboo-dev-cpp) [inline ui/imports/shared/status/StatusChatInput.qml:1149]: you commented this out right? should we create a bug for this if not already fixed?

File: ui/app/AppLayouts/Onboarding/views/LoginView.qml (hunk 1 above)
Comment (jrainville) [reply ui/app/AppLayouts/Onboarding/views/LoginView.qml:51]: 😱

File: ui/imports/shared/status/StatusChatInput.qml [hunk 4]
```diff
@@ -118,6 +122,21 @@ Rectangle {
         }
     }
 
+    Connections {
+        enabled: control.emojiPopupOpened
+        target: emojiPopup || null
```
Comment (Khushboo-dev-cpp) [inline ui/imports/shared/status/StatusChatInput.qml:127]: simple assigning emojiPopup would work?

File: ui/app/AppLayouts/Chat/views/CommunityColumnView.qml (hunk 2 above)
Comment (jrainville) [reply ui/app/AppLayouts/Chat/views/CommunityColumnView.qml:93]: The count was `undefined` sometimes and showing a warning

File: ui/imports/shared/status/StatusChatInput.qml (hunk 3 above)
Comment (jrainville) [reply ui/imports/shared/status/StatusChatInput.qml:1149]: My bad, I'll try to fix right now

File: ui/imports/shared/status/StatusChatInput.qml (hunk 4 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/status/StatusChatInput.qml:127]: if its same reason here we can do a null check? I feel this more difficult tor read

File: ui/app/AppLayouts/Chat/views/CommunityColumnView.qml (hunk 2 above)
Comment (jrainville) [reply ui/app/AppLayouts/Chat/views/CommunityColumnView.qml:93]: I'm going to have to disagree with your suggestion. It's pretty standard practice in JS to do `falsey || other` and your suggestion is just way longer. Unless you have very strong feelings about this, I'll leave it as is.

**Comment by jrainville** [comment]
@Khushboo-dev-cpp fixed your comments. Please note that I fixed a couple of other stuff, like moving the Emoji code to StatusQ.

**Review by anastasiyaig — APPROVED** [review]

**Comment by iurimatias** [comment]
@jrainville it needs another rebase, srry, afterwards feel free to merge it

**Comment by jrainville** [comment]
@iurimatias I need another approval to merge

**Review by iurimatias — APPROVED** [review]
